homeworks/08-functional-reactive-programming/src/index.ts

```diff
@@ -0,0 +1,73 @@
+import { defer, fromEvent, interval, of, Subscription, Observable } from "rxjs";
+import {
+  filter,
+  map,
+  tap,
+  pairwise,
+  startWith,
+  withLatestFrom,
+  mapTo,
+  scan,
+} from "rxjs/operators";
+import {
+  checkIfClicked,
+  drawImg,
+  generateBtn,
+  generateEmptyField,
+  genField,
+  getHappyCats,
+  resetGame,
+  score,
+} from "./renderer";
+
+import window from "./assets/window.png";
+import sadCat from "./assets/sad_cat.png";
+
+import "./index.css";
+
+const happyCats = getHappyCats();
+
+const windowImg = new Image();
+windowImg.src = window;
```

Comment:
Need to subscribe to `unload` event.

Reply:
Did you mean the `onload` event? 
I'm not sure how to do it in this case. 
Something like `sadCatImg.onload = () => sadCatImg.src = sadCat;` wouldn't work, because nothing is loading in the first place so the onload event would never trigger.
Or you meant to wait for the onload event before drawing like `sadCatImg.onload = () => drawImg(sadCatImg, coords);`, that also wouldn't work because when the `drawImg` function needs to be executed, the onload event is long passed.